manifests/pam.pp

```diff
@@ -1,374 +0,0 @@
-      fail('timelimit and timeout both set but do not match. These values must be the same.')
-    }
-    validate_integer($timelimit)
-  }
-  if !empty($timeout) {
-    if !empty($timelimit) and ($timeout != $timelimit) {
-      fail('timelimit and timeout both set but do not match. These values must be the same.')
-    }
-    validate_integer($timeout)
-  }
-  if !empty($bind_timelimit) {
-    if !empty($network_timeout) and ($network_timeout != $bind_timelimit) {
-      fail('bind_timelimit and network_timeout both set but do not match. These values must be the same.')
-    }
-    validate_integer($bind_timelimit)
-  }
-  if !empty($network_timeout) {
-    if !empty($bind_timelimit) and ($network_timeout != $bind_timelimit) {
-      fail('bind_timelimit and network_timeout both set but do not match. These values must be the same.')
-    }
-    validate_integer($network_timeout)
-  }
-  validate_bool($referrals)
-  validate_bool($restart)
-  validate_integer($idle_timelimit)
-  validate_bool($pam_lookup_policy)
-  validate_bool($pam_check_host_attr)
-  validate_bool($pam_check_service_attr)
-  validate_integer($pam_min_uid)
-  validate_integer($pam_max_uid)
-  validate_array_member($pam_password,[
-    'clear',
-    'clear_remove_old',
-    'crypt',
-    'nds',
-    'racf',
-    'ad',
-    'exop',
-    'exop_send_old'
-  ])
-  validate_bool($ssl)
-  validate_bool($start_tls)
-  validate_bool($tls_checkpeer)
-  if !empty($tls_cacertfile) { validate_absolute_path($tls_cacertfile) }
-  if !empty($tls_cacertdir) { validate_absolute_path($tls_cacertdir) }
-  validate_array($tls_ciphers)
-  if !empty($tls_cert) { validate_absolute_path($tls_cert) }
-  if !empty($tls_key) { validate_absolute_path($tls_key) }
-  validate_bool($tls_randfile)
-  validate_integer($threads)
-  validate_absolute_path($nslcd_conf_dir)
-  validate_integer($nslcd_uid)
-  validate_integer($nslcd_gid)
-
-
-  if $use_simp_pki {
-    include '::pki'
-  }
-
-  if empty($tls_cert) and $use_simp_pki {
-    $_tls_cert = "${nslcd_conf_dir}/pki/public/${::fqdn}.pub"
-  }
-  else {
-    $_tls_cert = $tls_cert
-  }
-
-  if empty($tls_key) and $use_simp_pki {
-    $_tls_key = "${nslcd_conf_dir}/pki/private/${::fqdn}.pem"
-  }
-  else {
-    $_tls_key = $tls_key
-  }
-
-  if empty($tls_cacertdir) and $use_simp_pki {
-    $_tls_cacertdir = "${nslcd_conf_dir}/pki/cacerts"
-  }
-  else {
-    $_tls_cacertdir = $tls_cacertdir
-  }
-
-  if empty($tls_cacertfile) and $use_simp_pki {
-    $_tls_cacertfile = "${nslcd_conf_dir}/pki/cacerts/cacerts.pem"
-  }
-  else {
-    $_tls_cacertfile = $tls_cacertfile
-  }
-
-  if $use_auditd {
-    include '::auditd'
-
-    auditd::add_rules { 'ldap.conf':
-      content => "-w ${ldap_conf} -p a -k CFG_etc_ldap",
-      require => File[$ldap_conf]
-    }
-  }
-
-  if $ssl and $use_simp_pki {
-    Class['pki'] -> File[$ldap_conf]
-  }
-
-  # Complete hackery to account for legacy upgrades.
-  if $use_sssd {
```

Comment:
Yes.  I asked if we needed to check for sssd and was told yes, someone might not want to use it, so leave the checks in.

I changed the name to sssd to match the global catalyst.